---
description: 编译器警告（等级 2）CS0162
title: 编译器警告（等级 2）CS0162
ms.date: 07/20/2015
f1_keywords:
- CS0162
helpviewer_keywords:
- CS0162
ms.assetid: 369b5b02-a9cc-404b-b758-4184285af2de
ms.openlocfilehash: 584faebca993de75c6e02f0b3d885fa547be8cea
ms.sourcegitcommit: 10e719780594efc781b15295e499c66f316068b8
ms.translationtype: MT
ms.contentlocale: zh-CN
ms.lasthandoff: 02/14/2021
ms.locfileid: "100424287"
---
# <a name="compiler-warning-level-2-cs0162"></a>编译器警告（等级 2）CS0162

检测到无法访问的代码
  
编译器检测到永远不执行的代码。

## <a name="example"></a>示例

下面的示例生成 CS0162：

```csharp
// CS0162.cs
// compile with: /W:2
public class Program
{
    public static void Main()
    {
        goto lab1;
        {
            // The following statements cannot be reached:
            int i = 9;   // CS0162
            i++;
        }
    lab1:
        {
        }
    }
}
```

生成此错误的另一个常见示例如下所示：

```csharp
public static class Class1
{
    public static string Method1()
    {
        string x = "a";
        switch (x)
        {
            case "a":
                return "a";
                break;          // CS0162
        }
        return "";
    }
}
```

`break`无法到达语句，因为它出现在语句之后 `return` 。
`return`语句结束封闭 `case` 分支。
